pkg/provider/provider.go

```diff
@@ -566,6 +579,102 @@ func (k *kubeProvider) StreamInvoke(
 				//
 
 				watch.Stop()
+				return nil
+			}
+		}
+	case streamInvokePodLogs:
+		//
+		// Set up log stream for Pod.
+		//
+
+		namespace := "default"
+		if args["namespace"].HasValue() {
+			namespace = args["namespace"].StringValue()
+		}
+
+		if !args["name"].HasValue() {
+			return fmt.Errorf(
+				"could not retrieve pod logs because the pod name was not present")
+		}
+		name := args["name"].StringValue()
+
+		podLogs, err := k.logClient.Logs(namespace, name)
+		if err != nil {
+			return err
+		}
+		defer podLogs.Close()
+
+		//
+		// Enumerate logs by line. Send back to the user.
+		//
+		// TODO: We send the logs back one-by-one, but we should probably batch them instead.
+		//
+
+		logLines := make(chan string)
+		defer close(logLines)
+		done := make(chan error)
+		defer close(done)
+
+		go func() {
+			podLogLines := bufio.NewScanner(podLogs)
+			for podLogLines.Scan() {
```

Comment:
From the `Scan` docs:
> Scan panics if the split function returns too many empty tokens without advancing the input. This is a common error mode for scanners.

Do we need to handle this with a `recover` or something?

Reply:
Should we? Generally in Go one doesn't try to recover from panics, except to relay information about a failure before shutting down gracefully.